checker/src/main/java/org/checkerframework/checker/determinism/DeterminismVisitor.java

```diff
@@ -488,40 +494,44 @@ public Void visitMethod(MethodTree node, Void p) {
                             .getAnnotationInHierarchy(atypeFactory.NONDET));
         }
         boolean isPolyPresent = false;
-        boolean isPolyUpPresent = false;
-        boolean isPolyDownPresent = false;
         boolean isPolyUsePresent = false;
-        boolean isPolyUpDetPresent = false;
         for (AnnotationMirror atm : polyAnnotations) {
             if (AnnotationUtils.areSame(atm, atypeFactory.POLYDET_UP)) {
-                isPolyUpPresent = true;
+                checker.report(Result.failure("invalid.polydet.up"), node);
```

Comment:
These errors say `invalid to write @PolyDet("up") without a corresponding @PolyDet`, but it should be something about writing `@PolyDet("up")` on a parameter is illegal.